error: ‘computeCollisions’ is not a member of ‘pinocchio’ [ros-humble]

Hello there,

I have encountered this issue when `colcon build` the project. I followed your instructions step by step.
```
root@hp-desktop:/workspace# colcon build
Starting >>> pinocchio_ros_example
--- stderr: pinocchio_ros_example                             
CMake Warning at /usr/share/cmake-3.22/Modules/FindBoost.cmake:2201 (message):
  No header defined for python310; skipping header check (note: header-only
  libraries have no designated component)
Call Stack (most recent call first):
  /opt/ros/humble/lib/x86_64-linux-gnu/cmake/eigenpy/boost.cmake:144 (find_package)
  /opt/ros/humble/lib/x86_64-linux-gnu/cmake/eigenpy/eigenpyConfig.cmake:149 (SEARCH_FOR_BOOST_PYTHON)
  /usr/share/cmake-3.22/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/ros/humble/lib/x86_64-linux-gnu/cmake/pinocchio/pinocchioConfig.cmake:161 (find_dependency)
  CMakeLists.txt:13 (find_package)

/workspace/src/pinocchio_ros_example/src/pinocchio_example.cpp: In function ‘int main(int, char**)’:
/workspace/src/pinocchio_ros_example/src/pinocchio_example.cpp:51:14: error: ‘computeCollisions’ is not a member of ‘pinocchio’; did you mean ‘ComputeCollision’?
   51 |   pinocchio::computeCollisions(model, data, collision_model, collision_data, q);
      |              ^~~~~~~~~~~~~~~~~
      |              ComputeCollision
gmake[2]: *** [CMakeFiles/pinocchio_example.dir/build.make:76: CMakeFiles/pinocchio_example.dir/src/pinocchio_example.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/pinocchio_example.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< pinocchio_ros_example [7.26s, exited with code 2]

Summary: 0 packages finished [7.35s]
  1 package failed: pinocchio_ros_example
  1 package had stderr output: pinocchio_ros_example
```

I have also tried the package in Wolf's github, same issue. 
I also did `colcon build` on my ros-humble (not in docker), both packages have the same issue.

Could you give some hints please? Many thanks.

I added 
`#include "pinocchio/fwd.hpp"` at the beginning in `pinocchio_example.cpp`, the CMake warning is gone.
But the error of `computCollisions is not a member of pinocchio` still exists.

```
root@hp-desktop:/workspace# colcon build
Starting >>> pinocchio_ros_example
--- stderr: pinocchio_ros_example                             
/workspace/src/pinocchio_ros_example/src/pinocchio_example.cpp: In function ‘int main(int, char**)’:
/workspace/src/pinocchio_ros_example/src/pinocchio_example.cpp:51:14: error: ‘computeCollisions’ is not a member of ‘pinocchio’; did you mean ‘ComputeCollision’?
   51 |   pinocchio::computeCollisions(model, data, collision_model, collision_data, q);
      |              ^~~~~~~~~~~~~~~~~
      |              ComputeCollision
gmake[2]: *** [CMakeFiles/pinocchio_example.dir/build.make:76: CMakeFiles/pinocchio_example.dir/src/pinocchio_example.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/pinocchio_example.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
---
Failed   <<< pinocchio_ros_example [6.19s, exited with code 2]

Summary: 0 packages finished [6.28s]
  1 package failed: pinocchio_ros_example
  1 package had stderr output: pinocchio_ros_example

```

--- a/.env
+++ b/.env
@@ -1,0 +1,2 @@
+# Environment variables to configure Docker builds.
+ROS_DISTRO=humble

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,8 +1,10 @@
-FROM ros:humble as base
+ARG ROS_DISTRO=humble
 
-RUN apt update && apt install -y ros-humble-pinocchio
+FROM ros:${ROS_DISTRO} as base
+
+RUN apt update && apt install -y ros-${ROS_DISTRO}-pinocchio
 
 RUN mkdir -p /workspace/src
 WORKDIR /workspace
 
-RUN echo "source /opt/ros/humble/setup.bash" >> ~/.bashrc
+RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash" >> ~/.bashrc

--- a/README.md
+++ b/README.md
@@ -3,11 +3,12 @@
 This is a basic example of building a ROS 2 package that uses the [Pinocchio](https://github.com/stack-of-tasks/pinocchio) library.
 
 > [!NOTE]
-> Collision checking support currently only works on ROS 2 Humble. See https://github.com/stack-of-tasks/pinocchio/issues/2504 for more information.
+> Collision checking temporarily does not work on ROS 2 Jazzy, until version 3.4.0 or higher is synced. See https://github.com/stack-of-tasks/pinocchio/issues/2504 for more information.
 
 To run this example:
 
 ```shell
+export ROS_DISTRO=humble  # or jazzy/rolling
 docker compose build
 docker compose run base
 ```

--- a/docker-compose.yaml
+++ b/docker-compose.yaml
@@ -1,11 +1,11 @@
 services:
   base:
-    image: pinocchio_ros_cpp:base
+    image: pinocchio_ros_cpp:${ROS_DISTRO}
     build:
       context: .
       dockerfile: Dockerfile
       args:
-        ROS_DISTRO: humble
+        ROS_DISTRO: ${ROS_DISTRO}
       target: base
     # Interactive shell
     stdin_open: true

--- a/pinocchio_ros_example/src/pinocchio_example.cpp
+++ b/pinocchio_ros_example/src/pinocchio_example.cpp
@@ -5,6 +5,7 @@
 #include "pinocchio/algorithm/geometry.hpp"
 #include "pinocchio/algorithm/jacobian.hpp"
 #include "pinocchio/algorithm/joint-configuration.hpp"
+#include "pinocchio/collision/broadphase.hpp"
 #include "pinocchio/parsers/urdf.hpp"
 #include "pinocchio/parsers/srdf.hpp"
  

